File: solutions/julia/nucleotide-count/1/nucleotide-count.jl
"""
    count_nucleotides(strand)

The count of each nucleotide within `strand` as a dictionary.

Invalid strands raise a `DomainError`.

"""
function count_nucleotides(strand)
    if match(r"[^ACGT]",strand)!=nothing    
        throw(DomainError("Invalid strand", strand))
    end
    mergewith(+,
        Dict('A'=>0, 'C'=>0, 'G'=>0, 'T'=>0),
        Dict(nucleotid => count( ==(nucleotid), strand) for nucleotid in strand))

end

File: solutions/javascript/nucleotide-count/1/nucleotide-count.js
//
// Nucleotid count

export function countNucleotides(strand) {
  let count = {'A':0, 'C':0, 'G':0, 'T':0};
  strand.split('').forEach(n => {if(!isNaN(count[n])){count[n] = count[n] + 1} else throw Error("Invalid nucleotide in strand")});
  return count['A'].toString() + ' '  + count['C'].toString() + ' ' + count['G'].toString() + ' ' + count['T'].toString();
}
